datafusion/expr/src/expr_schema.rs

```diff
@@ -142,31 +180,39 @@ impl ExprSchemable for Expr {
             | Expr::IsUnknown(_)
             | Expr::IsNotTrue(_)
             | Expr::IsNotFalse(_)
-            | Expr::IsNotUnknown(_) => Ok(DataType::Boolean),
+            | Expr::IsNotUnknown(_) => Ok(LogicalType::Boolean),
             Expr::ScalarSubquery(subquery) => {
                 Ok(subquery.subquery.schema().field(0).data_type().clone())
             }
             Expr::BinaryExpr(BinaryExpr {
                 ref left,
                 ref right,
                 ref op,
-            }) => get_result_type(&left.get_type(schema)?, op, &right.get_type(schema)?),
-            Expr::Like { .. } | Expr::SimilarTo { .. } => Ok(DataType::Boolean),
+            }) => {
+                // TODO not convert to physical DataType
+                let physical_type = get_result_type(
```

Comment:
TODO: Modify the function `get_result_type` to accept LogicalType as parameters.